plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/encryption/SSEHeaders.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spooling.filesystem.encryption;
+
+import io.trino.filesystem.Location;
+import io.trino.filesystem.encryption.EncryptionKey;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public interface SSEHeaders
+{
+    EncryptionKey extract(Map<String, List<String>> headers);
+
+    Map<String, List<String>> toHeaders(EncryptionKey encryption);
+
+    static SSEHeaders getSSEHeaders(Location location)
+    {
+        requireNonNull(location, "location is null");
+        return location.scheme()
+                .map(SSEHeaders::forScheme)
+                .orElseThrow(() -> new IllegalArgumentException("Unknown location scheme: " + location));
+    }
+
+    private static SSEHeaders forScheme(String scheme)
+    {
+        // These should match schemes supported in the FileSystemSpoolingModule
+        return switch (scheme) {
+            case "s3" -> new S3SSEHeaders();
+            case "gs" -> new GcsSSEHeaders();
+            case "abfs" -> new AzureSSEHeaders();
```

Comment:
Are those only schemes supported. There used to be `s3n` etc... in the past.

Also should those be exposed by TrinoFilesystem implementations and we should just reference those here?